clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -409,6 +414,20 @@ public synchronized void removeStorageEngine(String kafkaTopic) {
     factory.removeStorageEngine(storageEngine);
   }
 
+  AbstractStorageEngine openStorageEngine(String storeName) {
+    VeniceStoreVersionConfig storeConfig = configLoader.getStoreConfig(storeName);
+    storeConfig.setRestoreDataPartitions(false);
```

Comment:
If we want to drop all data partition in II case, then we need to setRestoreDataPartition to true. 

Also, looks like the test does not cover this.